### ISSUE TITLE
The value of the certificate in service client_certificate does not take effect after modification

### ISSUE THREAD

**Issue by Murphy-hub** [Kong/kong#9634]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Kong version (`$ kong version`)

2.8.1

### Current Behavior

I configured a service and set client_certificate. When I modify the actual certificate value of this certificate through admin Api, the client certificate provided by the service to the upstream is still the original certificate, and it is not updated to the value I modified.

#### Service

<img width="732" alt="image" src="https://user-images.githubusercontent.com/52234994/198829643-ffebeac4-e9d6-429a-b024-393d3521fff4.png">

#### Operate

Update Certificate content.

### Expected Behavior

It is hoped that when the content of the certificate is changed, the service can automatically obtain the latest certificate

### Steps To Reproduce

```markdown
k8s
```

### Anything else?

none

**Comment by StarlightIbuki** [comment]
Could you share the setup you are using, for example, hybrid or traditional mode, DBLess or with a DB?
After waiting for a period of time, is the cert still not updated?

**Comment by Murphy-hub** [comment]
@StarlightIbuki Thank you for your reply.
this is my helm configuration, is the traditional mode

```
  migrations:
    init: false
    preUpgrade: false
    postUpgrade: false
  deployment:
    daemonset: true
    hostNetwork: true
  env:
    log_level: info
    database: "postgres"
    pg_host: xxxxxx
    pg_name: kong
    pg_password: xxx
    pg_port: 1921
    pg_user: xxx
    prefix: "/kong"
    plugins: "bundled,simple-basic-auth"
    dns_resolver: 169.254.25.10
    error_default_type: "application/json"
    nginx_worker_processes: "auto"
    nginx_http_client_body_buffer_size: "4m"
  admin:
    enabled: true
    type: ClusterIP
    http:
      enabled: true
    tls:
      enabled: false
  proxy:
    enabled: true
    type: ClusterIP
    http:
      enabled: true
      servicePort: 80
      containerPort: 80
    tls:
      enabled: true
      servicePort: 443
      containerPort: 443
    stream:
      - containerPort: 7000
        servicePort: 7000
        protocol: TCP
      - containerPort: 9898
        servicePort: 9898
        protocol: TCP
      - containerPort: 7443
        servicePort: 7443
        protocol: TCP
        parameters:
          - ssl
          - reuseport
          - backlog=16384
```

**Comment by Murphy-hub** [comment]
> Could you share the setup you are using, for example, hybrid or traditional mode, DBLess or with a DB? After waiting for a period of time, is the cert still not updated?

The id value in the client_certificate in the Service object is always the same. I modified the actual content of the certificate through the management API, and it still hasn't been updated after a period of time.
And I have deployed the same function in other clusters. The certificate has been updated for many days, but the certificate obtained by the tour is still the old certificate content.

**Comment by Murphy-hub** [comment]
> 

I can only restart the kong pod, the certificate content will load normally.

**Comment by StarlightIbuki** [comment]
This sounds like a cache invalidation problem. We will address that soon.

**Comment by mayocream** [comment]
Hi @Murphy-hub, 

The issue you mentioned should be resolved by https://github.com/Kong/kong/pull/8934, the patch was included in the 3.0.0 release. You could try to patch the older kong versions with that fix or upgrade to 3.0.

**Comment by Murphy-hub** [comment]
@mayocream Nice, thank you very much
